cylc/flow/task_pool.py

```diff
@@ -913,25 +911,7 @@ def set_max_future_offset(self):
         if max_offset != orig and self.compute_runahead(force=True):
             self.release_runahead_tasks()
 
-    def set_do_reload(self, config: 'WorkflowConfig') -> None:
-        """Set the task pool to reload mode."""
-        self.config = config
-        self.stop_point = config.stop_point or config.final_point
-        self.do_reload = True
-
-        # find any old tasks that have been removed from the workflow
-        old_task_name_list = self.task_name_list
-        self.task_name_list = self.config.get_task_name_list()
-        for name in old_task_name_list:
-            if name not in self.task_name_list:
-                self.orphans.append(name)
-        for name in self.task_name_list:
-            if name in self.orphans:
-                self.orphans.remove(name)
```

Comment:
The list of orphans used to be persisted between reloads. Newly bereaved tasks were added to the list and only removed if a matching task was added back in, otherwise they were left in libo. However, they could be pulled back by a subsequent reload if the task was added back into the definition.

I think this is another piece of Cylc 7 hangover as at Cylc 8, orphans are swiftly destroyed unless ready to run.